Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/impl/CertificateManagerServiceImpl.java

```diff
@@ -210,34 +235,18 @@ private static void closeFiles(FileInputStream is, FileOutputStream os) {
             LOG.error("Unable to close File Stream: {}", ex.getLocalizedMessage(), ex);
         }
     }
-
-    private static boolean addCertificateToTrustStore(String storeType, String storeLoc, String passkey,
-        Certificate cert) throws CertificateManagerException {
-        FileInputStream is = null;
-        FileOutputStream os = null;
-        boolean isImportSuccessful = false;
-        try {
-            KeyStore secretStore = KeyStore.getInstance(storeType);
-            if (!PKCS11_TYPE.equalsIgnoreCase(storeType)) {
-                is = new FileInputStream(storeLoc);
-            }
-            secretStore.load(is, passkey.toCharArray());
-            secretStore.setCertificateEntry(cert.getAlias(), cert.getX509Cert());
-            os = new FileOutputStream(storeLoc);
-            secretStore.store(os, passkey.toCharArray());
-            isImportSuccessful = true;
-        } catch (final IOException | NoSuchAlgorithmException | CertificateException | KeyStoreException ex) {
-            LOG.error("Unable to add the Certifiate: ", ex.getLocalizedMessage(), ex);
-            throw new CertificateManagerException(ex.getMessage(), ex);
-        } finally {
-            closeFiles(is, os);
-        }
-        return isImportSuccessful;
+    
+    private boolean addCertificateToTrustStore(Certificate cert) throws CertificateManagerException, Exception {
+        
+        ImportCertificateRequestMessageType requestMessage = createImportCertRequest(cert);
+        ImportCertificateResponseMessageType response = (ImportCertificateResponseMessageType) getClient().invokePort(EntityConfigAdminPortType.class, "importCertificate", requestMessage);
```

Comment:
If possible, catch Exception and re-throw CertificateManagerException so that we don't need to throw general Exception on caller class?